gammapy/image/maps.py

```diff
@@ -260,7 +260,7 @@ def solid_angle(self):
         """
         Solid angle image
         """
-        xsky, ysky = self.coordinates(mode='edges')
+        xsky, ysky = self.coordinates(coord_type='galactic', mode='edges')
         omega = -np.diff(xsky, axis=1)[1:, :] * np.diff(ysky, axis=0)[:, 1:]
```

Comment:
Just to confirm what @adonath  is suggesting: just call `np.abs` to make it work for all images. Some images will have positive or negative values for CDELT and the easiest way to handle all images is to take the abs value.